engines/griffon/dialogs.cpp

```diff
@@ -0,0 +1,761 @@
+					if (mode == 1)
+						exitTitle = true;
+					break;
+				case Common::KEYCODE_UP:
+					cursel--;
+					if (cursel < 0)
+						cursel = (mode == 1 ? 3 : 2);
+					break;
+				case Common::KEYCODE_DOWN:
+					cursel++;
+					if (cursel >= (mode == 1 ? 4 : 3))
+						cursel = 0;
+					break;
+				case Common::KEYCODE_RETURN:
+					switch(cursel) {
+					case 0:
+						_ticks = g_system->getMillis();
+						ticks1 = _ticks;
+
+						exitTitle = true;
+						break;
+					case 1:
+						configMenu();
+						_ticks = g_system->getMillis();
+						ticks1 = _ticks;
+						break;
+					case 2:
+						_shouldQuit = true;
+						break;
+					case 3:
+						exitTitle = true;
+					default:
+						break;
+					}
+					break;
+				default:
+					break;
+				}
+			}
+		}
+
+		g_system->delayMillis(10);
+	} while (!_shouldQuit && !exitTitle);
+
+	_itemTicks = _ticks + 210;
+
+	if (config.music) {
+		haltSoundChannel(_menuChannel);
+		resumeSoundChannel(_musicChannel);
+		setChannelVolume(_musicChannel, config.musicVol);
+		_pmenu = false;
+	}
+}
+
+void GriffonEngine::configMenu() {
+	static const char *optionTitles[22] = {
+		"", "",
+		"", "", "", "",
+		"", "", "",
+		"Music:", "", "",
+		"Sound Effects:", "", "",
+		"Music Volume:", "",
+		"Effects Volume:", "", "", "", ""
+	};
+	static const char *optionValues[22] = {
+		"", "",
+		"", "", "", "",
+		"", "", "",
+		"On", "Off", "",
+		"On", "Off", "",
+		"[----------]", "",
+		"[----------]", "",
+		"Exit + Save", "",
+		"Exit"
+	};
+
+	int cursel = MINCURSEL;
+	bool exitMenu = false;
+
+	_ticks = g_system->getMillis();
+
+	Graphics::TransparentSurface *configwindow = loadImage("art/configwindow.bmp", true);
+	configwindow->setAlpha(160, true);
+
+	int ticks1 = _ticks;
+
+	_cloudImg->setAlpha(128, true);
+
+	do {
+		_videoBuffer->fillRect(Common::Rect(0, 0, _videoBuffer->w, _videoBuffer->h), 0);
+		_videoBuffer2->fillRect(Common::Rect(0, 0, _videoBuffer2->w, _videoBuffer2->h), 0);
+
+		rcDest.left = 256 + 256 * cos(PI / 180 * _cloudAngle * 40);
+		rcDest.top = 192 + 192 * sin(PI / 180 * _cloudAngle * 40);
+		rcDest.setWidth(320);
+		rcDest.setHeight(240);
+
+		_cloudImg->blit(*_videoBuffer, 0, 0, Graphics::FLIP_NONE, &rcDest);
+
+		rcDest.left = 256;
+		rcDest.top = 192;
+		rcDest.setWidth(320);
+		rcDest.setHeight(240);
+
+		_cloudImg->blit(*_videoBuffer, 0, 0, Graphics::FLIP_NONE, &rcDest);
+
+		configwindow->blit(*_videoBuffer);
+
+		int sy = SY;
+
+		for (int i = 0; i <= 21; i++) {
+			static char line[24];
+
+			int destColumn = 3;
+			if (i == 9 && config.music)
+				destColumn = 0;
+			else if (i == 10 && !config.music)
+				destColumn = 0;
+			else if (i == 12 && config.effects)
+				destColumn = 0;
+			else if (i == 13 && !config.effects)
+				destColumn = 0;
+			else if (i == 15 || i == 17) {
+				int vol = (i == 15 ? config.musicVol : config.effectsVol) * 9 / 255;
+				vol = CLIP(vol, 0, 9);
+
+				strcpy(line, "[----------]");
+				line[vol + 1] = 'X';
+				optionValues[i] = line;
```

Comment:
This looks suspect.. a single static line is being assigned to different option value indexes. Is this okay?

Reply:
Yes, it is a visual indicator:

<img width="752" alt="Screenshot 2019-11-07 at 00 22 45" src="https://user-images.githubusercontent.com/365188/68346565-bceeac80-00f4-11ea-9a1a-720f93eeba79.png">